### PR TITLE
Refine Appointment Booking: Add Form Validation and Duration-Based Amount Calculation

### LINKED ISSUE #202: Feature: Refine book appointment
### What does this issue do ?
This issue refine the process of booking an appointment by adding form handler , which prevent a user to navigate to another screen without fill required data.

### PR DESCRIPTION
**Pull request by KeynesBizimana**
### What does this pull request do?
Refines the appointment booking process by adding form validation handlers to ensure all required data is selected before navigation and introduces a duration variable to calculate the total amount.

### Description
Implemented the following functionalities:

- [ ] Form Validation Handlers: Ensured users cannot navigate to the next screen without selecting each required piece of information (gender, age, and problem description). This prevents incomplete data from being submitted and enhances user experience.

- [ ] Duration Variable for Amount Calculation: Added a variable to capture the duration a patient has selected. This duration is now used to accurately calculate the total amount for the appointment, ensuring that users are aware of the cost based on the duration of their selected appointment.

- [ ] Supabase Integration: The selected data (gender, age, problem description, and duration) is managed and stored through Supabase, maintaining consistency and reliability in the appointment booking process.

### Screenshots

Screenshot_20240710_144720_Expo Go
Screenshot_20240710_144658_Expo Go

### CODE REVIEW

File: app/(app)/ActionMenu/Booking/Patient-details.tsx [hunk 1]
```diff
@@ -70,10 +80,34 @@ const PatientDetails = () => {
   }, [loggeduser])
   console.log("this is profile:",profile)
   console.log("this is logged user:", loggeduser)
-  console.log("this is patient_id:",patient_id)
+  console.log("this is patient_id:", patient_id)
+
```
Comment (Irirwanirira) [inline app/(app)/ActionMenu/Booking/Patient-details.tsx]: Remove these console.log to have a clean code base.

File: app/(app)/ActionMenu/Booking/Patient-details.tsx [hunk 2]
```diff
@@ -70,10 +80,34 @@ const PatientDetails = () => {
...
+    const monthDifference = today.getMonth() - birthDate.getMonth();
+    if (monthDifference < 0 || (monthDifference === 0 && today.getDate() < birthDate.getDate())) {
+      return age - 1;
+    }
+    return age;
+  };
   const handleInputChange = (name: string, value: string) => {
     setProfile({ ...profile, [name]: value });
   }
+  console.log("this is default value:",profile?.gender,profile?.age)
```
Comment (Irirwanirira) [inline app/(app)/ActionMenu/Booking/Patient-details.tsx]: same here, they may be for testing purpose but not good on develop

File: app/(app)/ActionMenu/Booking/Patient-details.tsx [hunk 3]
```diff
@@ -124,10 +158,9 @@ const PatientDetails = () => {
           <DropDown
             data={[
                { value: `${profile?.gender}`, label:`${profile?.gender} ` },
-              { value: "Male", label: "Male" },
-              { value: "Female", label: "Female" },
             ]}
-            defaultvalue={profile?.Gender}
+            defaultValue={selectedGender}
+            onSelect={(value)=>setSelectedGender(value)}
           />
```
Comment (Irirwanirira) [inline app/(app)/ActionMenu/Booking/Patient-details.tsx]: kindly for the user experience as we are fetching data form the data base and they don't need to be edited at this stage consider displaying them in the <Text> tag or disable <Input>

File: app/(app)/ActionMenu/Booking/Patient-details.tsx [hunk 4]
```diff
@@ -148,11 +181,11 @@ const PatientDetails = () => {
           <DropDown
             data={[
               { value: `${profile?.age} Years`, label:`${profile?.age} Years` },
-              { value: "27 years", label: "27 years" },
-              { value: "28 years", label: "28 years" },
+
```
Comment (Irirwanirira) [inline app/(app)/ActionMenu/Booking/Patient-details.tsx]: Kindly consider displaying the following not in the Drop down

File: app/(app)/ActionMenu/Booking/Patient-details.tsx [hunk 5]
```diff
@@ -182,9 +215,10 @@ const PatientDetails = () => {
         >
           <Button
             title="Next"
-            onPress={() => {
-              router.push({ pathname: "/(app)/ActionMenu/Booking/SelectPayment",params:{doctor_id:Doctor_id,hour:hour,date:date,packageTitle:packageTitle,packagePrice:packagePrice,problem:text,user_id:loggeduser,patient_id:patient_id} });
-            }}
+            onPress={handleNextPress}
+            // onPress={() => {
+            //   router.push({ pathname: "/(app)/ActionMenu/Booking/SelectPayment",params:{doctor_id:Doctor_id,hour:hour,date:date,packageTitle:packageTitle,packagePrice:packagePrice,problem:text,user_id:loggeduser,patient_id:patient_id,duration:duration} });
+            // }}
```
Comment (Irirwanirira) [inline app/(app)/ActionMenu/Booking/Patient-details.tsx:210]: Kindly remove the commented code to have a clean codebase

**Review by Irirwanirira — CHANGES_REQUESTED** [review]
Good job on implementing all this work, but below are some changes requested, kindly refer to the comments below and re-request a review after working in them. 

-  After the user presses on the success button in the modal he is redirected to the lists of appointments which is very good but whenever he gets back to the home screen ... they give him a page to add his pin again which triggers the modal for the second time. and from here a user can book multiple appointments at once.  please work on that.

File: app/(app)/ActionMenu/Booking/Patient-details.tsx (hunk 1 above)
Comment (KeynesBizimana) [reply app/(app)/ActionMenu/Booking/Patient-details.tsx]: Done

File: app/(app)/ActionMenu/Booking/Patient-details.tsx (hunk 2 above)
Comment (KeynesBizimana) [reply app/(app)/ActionMenu/Booking/Patient-details.tsx]: Done

File: app/(app)/ActionMenu/Booking/Patient-details.tsx (hunk 3 above)
Comment (KeynesBizimana) [reply app/(app)/ActionMenu/Booking/Patient-details.tsx]: Done

File: app/(app)/ActionMenu/Booking/Patient-details.tsx (hunk 4 above)
Comment (KeynesBizimana) [reply app/(app)/ActionMenu/Booking/Patient-details.tsx]: Done

File: app/(app)/ActionMenu/Booking/Patient-details.tsx (hunk 5 above)
Comment (KeynesBizimana) [reply app/(app)/ActionMenu/Booking/Patient-details.tsx:210]: Done

**Review by munezeromicha — APPROVED** [review]

**Review by mr3nz1 — APPROVED** [review]

**Review by Nkbtemmy — APPROVED** [review]

**Pull request merged**